src/tools/compiletest/src/main.rs

```diff
@@ -597,7 +631,7 @@ fn collect_tests_from_dir(
         let file = file?;
         let file_path = file.path();
         let file_name = file.file_name();
-        if is_test(&file_name) {
+        if is_test(&file_name) && (!config.only_modified || only_modified.contains(&file_path)) {
```

Comment:
This means that if the `modified_test` functions errors (which is signalled by returning an empty `Vec`), then no tests are run, or am I wrong?
The old way of using `Option` was good.

Reply:
yes, the old way has a bug for the `relative_dir`.
Old way also won't run any test if it specified with `only-modified` and `git` run with error.

Reply:
Then now what is the difference between "No tests modified/added" and "An error occured"?

Reply:
Just added error message for error occured.